decode_test.go

```diff
@@ -41,46 +76,67 @@ func TestDecoder_scan(t *testing.T) {
 			}},
 		},
 		{
-			"y=f\ny=g",
-			[][]kv{
+			data: "y=f\ny=g",
+			dec:  defaultDecoder,
+			want: [][]kv{
 				{{[]byte("y"), []byte("f")}},
 				{{[]byte("y"), []byte("g")}},
 			},
 		},
 		{
-			"y=f  \n\x1e y=g",
-			[][]kv{
+			data: "y=f  \n\x1e y=g",
+			dec:  defaultDecoder,
+			want: [][]kv{
 				{{[]byte("y"), []byte("f")}},
 				{{[]byte("y"), []byte("g")}},
 			},
 		},
 		{
-			"y= d y=g",
-			[][]kv{{
+			data: "y= d y=g",
+			dec:  defaultDecoder,
+			want: [][]kv{{
 				{[]byte("y"), nil},
 				{[]byte("d"), nil},
 				{[]byte("y"), []byte("g")},
 			}},
 		},
 		{
-			"y=\"f\"\ny=g",
-			[][]kv{
+			data: "y=\"f\"\ny=g",
+			dec:  defaultDecoder,
+			want: [][]kv{
 				{{[]byte("y"), []byte("f")}},
 				{{[]byte("y"), []byte("g")}},
 			},
 		},
 		{
-			"y=\"f\\n\"y=g",
-			[][]kv{{
+			data: "y=\"f\\n\"y=g",
+			dec:  defaultDecoder,
+			want: [][]kv{{
 				{[]byte("y"), []byte("f\n")},
 				{[]byte("y"), []byte("g")},
 			}},
 		},
+		{
+			data: strings.Repeat(`y=f `, 5),
+			dec:  func(s string) *Decoder { return NewDecoderSize(strings.NewReader(s), 21) },
+			want: [][]kv{{
+				{[]byte("y"), []byte("f")},
+				{[]byte("y"), []byte("f")},
+				{[]byte("y"), []byte("f")},
+				{[]byte("y"), []byte("f")},
+				{[]byte("y"), []byte("f")},
+			}},
+		},
+		{
+			data: strings.Repeat(`y=f `, 5),
+			dec:  func(s string) *Decoder { return NewDecoderSize(strings.NewReader(s), 20) },
+			want: [][]kv{},
+		},
```

Comment:
This test case causes a failure.

```
$ go test
--- FAIL: TestDecoder_scan (0.00s)
    decode_test.go:153: got err: bufio.Scanner: token too long
    decode_test.go:156:
          in: "y=f y=f y=f y=f y=f "
         got: []
        want: []
FAIL
exit status 1
FAIL    github.com/go-logfmt/logfmt     0.154s
```

It looks like it is intended to trigger the error, but the test loop doesn't know that's what we're expecting. I think this case belongs in `TestDecoder_errors` below.

Reply:
Ah, I didn't see that function, cheers!